src/parsers/forecast_parser.jl

```diff
@@ -308,20 +326,21 @@ function clear_forecasts!(sys::System)
 end
 
 """
-    split_forecasts!(sys::System, 
-                    forecasts, 
-                    interval::Dates.Period, 
+    split_forecasts!(sys::System,
+                    forecasts,
+                    interval::Dates.Period,
                     horizon::Int) where T <: Forecast
 
-Replaces system forecasts with a set of forecasts by incrementing through an iterable 
+Replaces system forecasts with a set of forecasts by incrementing through an iterable
 set of forecasts by interval and horizon.
 
 """
 function split_forecasts!(sys::System,
-                         forecasts, # must be an iterable
+                         forecasts::FlattenIteratorWrapper{T}, # must be an iterable
                          interval::Dates.Period,
                          horizon::Int) where T <: Forecast
 
+    isempty(forecasts) && throw(ArgumentError, "Forecasts is empty")
```

Comment:
@jd-lara  Just FYI, this is incorrect.  I'll fix it in my current branch.

`throw(ArgumentError("Forecasts is empty"))`